plutus-contract/src/Plutus/Contract/Test/ContractModel.hs

```diff
@@ -1134,3 +1143,35 @@ checkNoLockedFundsProof options spec NoLockedFundsProof{nlfpMainStrategy   = mai
                               "  " ++ show bal'
                     DL.assert err (bal `leq` bal')
 
+
+-- | Check that running a contract model does not result in validation
+-- failures due to partial functions from the prelude being used in
+-- a validator.
+checkNoPartiality :: ContractModel m
+                  => [ContractInstanceSpec m]
+                  -> Property
+checkNoPartiality = checkNoPartialityWithOptions defaultCheckOptions
+
+-- | Check that running a contract model does not result in validation
+-- failures due to partial functions from the prelude being used in
+-- a validator.
+checkNoPartialityWithOptions :: forall m. ContractModel m
+                             => CheckOptions
+                             -> [ContractInstanceSpec m]
+                             -> Property
+checkNoPartialityWithOptions opts handleSpecs = forAllShrink arbitrary shrink $ \ acts -> go check acts
+  where
+    check = checkOnchain .&&. (assertNoFailedTransactions .||. checkOffchain)
+    checkOnchain = assertChainEvents checkEvents
+
+    checkOffchain = assertFailedTransaction (\ _ _ -> all (either checkEvent (const True) . sveResult))
+
+    checkEvent (EvaluationError [] "CekEvaluationFailure")  = False
+    checkEvent (EvaluationError log "CekEvaluationFailure") = Test.pack (Builtins.fromBuiltin errorTag) `Text.isPrefixOf` last log
+    checkEvent _                                            = True
```

Comment:
There will also be errors that arise from a partial builtin failing. Those will look different, they should look like `BuiltinErrorCall` or something. The easiest way to trigger one would be to pass a malformed `Data` to a validator; `unsafeFromData` will fail in this way.